executor/grant_test.go

```diff
@@ -474,6 +474,37 @@ func TestGrantOnNonExistTable(t *testing.T) {
 	require.NoError(t, err)
 	_, err = tk.Exec("grant Select,Update on test.xx to 'genius'")
 	require.NoError(t, err)
+
+	// issue #29268
+	tk.MustExec("CREATE DATABASE d29268")
+	defer tk.MustExec("DROP DATABASE IF EXISTS d29268")
+	tk.MustExec("USE d29268")
+	tk.MustExec("CREATE USER u29268")
+	defer tk.MustExec("DROP USER u29268")
+
+	// without create privilege
+	err = tk.ExecToErr("GRANT SELECT ON t29268 TO u29268")
+	require.Error(t, err)
+	require.True(t, terror.ErrorEqual(err, infoschema.ErrTableNotExists))
+	err = tk.ExecToErr("GRANT UPDATE ON t29268 TO u29268")
+	require.Error(t, err)
+	require.True(t, terror.ErrorEqual(err, infoschema.ErrTableNotExists))
+	err = tk.ExecToErr("GRANT DROP, DELETE ON t29268 TO u29268")
+	require.Error(t, err)
+	require.True(t, terror.ErrorEqual(err, infoschema.ErrTableNotExists))
+	err = tk.ExecToErr("GRANT CREATE VIEW, SHOW VIEW ON t29268 TO u29268")
+	require.Error(t, err)
+	require.True(t, terror.ErrorEqual(err, infoschema.ErrTableNotExists))
+	err = tk.ExecToErr("GRANT INSERT, REFERENCES, ALTER ON t29268 TO u29268")
+	require.Error(t, err)
+	require.True(t, terror.ErrorEqual(err, infoschema.ErrTableNotExists))
+
+	// with create privilege
+	tk.MustExec("GRANT CREATE ON t29268 TO u29268")
+	tk.MustExec("GRANT CREATE, SELECT ON t29268 TO u29268")
+	tk.MustExec("GRANT CREATE, DROP, CREATE VIEW ON t29268 TO u29268")
+	tk.MustExec("GRANT CREATE, INSERT, REFERENCES, ALTER ON t29268 TO u29268")
```

Comment:
Are all the privileges take effect after the table is created? Please try it in MySQL and add tests here.

Reply:
Yes, it works after creation. Test cases added.